mobile/lib/modules/asset_viewer/ui/description_input.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/shared/models/asset.dart';
+import 'package:immich_mobile/shared/models/exif_info.dart';
+import 'package:immich_mobile/shared/providers/api.provider.dart';
+import 'package:immich_mobile/shared/providers/db.provider.dart';
+import 'package:logging/logging.dart';
+import 'package:openapi/api.dart';
+
+class DescriptionInput extends HookConsumerWidget {
+  DescriptionInput({
+    super.key,
+    required this.asset,
+  });
+
+  final Asset asset;
+  final Logger _log = Logger('DescriptionInput');
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    var isDarkTheme = Theme.of(context).brightness == Brightness.dark;
+    var textColor = isDarkTheme ? Colors.white : Colors.black;
+    var controller = useTextEditingController();
+    var db = ref.watch(dbProvider);
+    var focusNode = useFocusNode();
+    var isFocus = useState(false);
+    var isTextEmpty = useState(controller.text.isEmpty);
+
+    getLatestDescription() async {
+      var latestAssetFromServer = await ref
+          .watch(apiServiceProvider)
+          .assetApi
+          .getAssetById(asset.remoteId!);
+      var localExifInfo = await db.exifInfos.get(asset.exifInfo!.id!);
+
+      if (latestAssetFromServer != null && localExifInfo != null) {
+        controller.text = latestAssetFromServer.exifInfo?.description ?? '';
+        localExifInfo.description = latestAssetFromServer.exifInfo!.description;
+
+        await db.writeTxn(
+          () => db.exifInfos.put(localExifInfo),
+        );
+      }
+    }
+
+    useEffect(
+      () {
+        getLatestDescription();
+        return null;
+      },
+      [],
+    );
+
+    submitDescription(String description) async {
+      try {
+        var result = await ref.watch(apiServiceProvider).assetApi.updateAsset(
+              asset.remoteId!,
+              UpdateAssetDto(description: description),
+            );
+
+        if (result?.exifInfo?.description != null) {
+          controller.text = result!.exifInfo!.description!;
+          var exifInfo = await db.exifInfos.get(asset.exifInfo!.id!);
+
+          if (exifInfo != null) {
+            exifInfo.description = result.exifInfo!.description;
+            await db.writeTxn(
+              () => db.exifInfos.put(exifInfo),
+            );
+          }
+        }
+      } catch (e) {
+        _log.severe("Error updating description", e);
+        ScaffoldMessenger.of(context).showSnackBar(
+          const SnackBar(
+            content: Text(
+              'Failed to update description! Check log for more details',
+            ),
+          ),
+        );
+      }
+    }
+
+    return TextField(
+      focusNode: focusNode,
+      onTap: () => isFocus.value = true,
+      onChanged: (value) {
+        isTextEmpty.value = false;
+      },
+      onTapOutside: (a) async {
```

Comment:
I've never used this before, but it looks better than using listeners on the controller. Neat!